enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,168 @@
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure. 
+
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
+The OCM API provides an endpoint that will serve the complete `pull-secret` definition when the migration window is opened. 
+The migration window will be opened for certain period of time (e.g 5 days) so tha the Insights Operator can safely update the secret.
+
+Open question: What exactly will the API return outside of the given migration window?  
+
+### Monitoring
+
+There will be a new metric registered by the Insights Operator, which will count the number of responses (according the HTTP response status code) from the OCM API.
+
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify that the Insights Operator is able to download and update the `pull-secret` content
+
+### Graduation Criteria
+
+Thera are no graduation criteria. This feature is planned as a fast track feature.
+
+#### Dev Preview -> Tech Preview
+
+Empty
+
+#### Tech Preview -> GA
+
+Empty
+
+#### Removing a deprecated feature
+
+Removing this feature will require updating the Insights operator code base.
+
+### Upgrade / Downgrade Strategy
+
+There is no upgrade/downgrade strategy needed.
+
+### Version Skew Strategy
+
+There is no Skew strategy needed. This work should have no impact on the upgrade. It doesn't require any coordinated behavior in the control plane. No other components will change.
+
+
+### Operational Aspects of API Extensions
+
+No API extension.
+
+#### Failure Modes
+
+TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+There are no major milestones in the implementation history.
+
+## Drawbacks
+
+The performance of the OCM API can be a possible drawback.
```

Comment:
How often is Insights expected to poll the API?  If it's hourly or slower, it's hard to imagine this being a significant load on the OCM service side.  Are there latency constraints on the UX that make hourly polling for slow responses too slow?  Worth talking about latency constraints or lack thereof in the enhancement?

Reply:
Well we have been talking about daily interval AFAIK..... but @abhgupta can share his ideas/concerns

Reply:
This API call will be for exactly 1 row in 1 table with a well-indexed ID field.  It returns an instance of "ClusterTransfer" with details populated or it's empty.  Either way, it should be very fast. This is a scalable solution.

Reply:
I updated this section.